src/Raven.Server/Documents/Indexes/Persistence/Lucene/CustomSerialMergeScheduler.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Threading;
+using Lucene.Net.Index;
+using Lucene.Net.Store;
+using Raven.Client.Documents.Indexes;
+using Raven.Server.Utils;
+
+namespace Raven.Server.Documents.Indexes.Persistence.Lucene
+{
+    public class CustomSerialMergeScheduler : SerialMergeScheduler
+    {
+        private IndexingStatsScope _commitStats;
+
+        public void SetCommitStats(IndexingStatsScope commitStats)
+        {
+            _commitStats = commitStats;
+        }
+
+        public override void Merge(IndexWriter writer, IState state)
+        {
+            using (_commitStats?.For(IndexingOperation.Lucene.Merge))
+            {
+                base.Merge(writer, state);
```

Comment:
Can we provide additional information here? In particular, what was merged, context, etc?
https://github.com/ravendb/lucenenet/blob/3.0.3-netcore/src/Lucene.Net/Index/MergePolicy.cs#L72

Reply:
Number of items that are being merged is also interesting
Also, I think we should be able to set a time limit, no?
So we'll only allow a certain amount of time to go here, then force a merge if we need to when the index is idle?

Reply:
We have IndexingRunStats so it should not be a problem. We can just copy implementation of SerialMergeScheduler (https://github.com/ravendb/lucenenet/blob/3.0.3-netcore/src/Lucene.Net/Index/SerialMergeScheduler.cs) and write our own that will accumulate this.

What would be a reasonable time limit? What is the impact of not doing merges?

Reply:
Setting a time limit doesn't improve performance. It just delays the merge to the next batch.
I added the merge details.